### PR TITLE
chore(replica_calculator): use WaitForCacheSync after sharedInformerFactory Start in unit test

### PR DESCRIPTION
**Pull request by omerap12**
#### What type of PR is this?
/kind cleanup

#### Which issue(s) this PR fixes:

Part of: #126256 

#### Special notes for your reviewer:
As requested in issue #126256 , I'm splitting this into separate PRs.
#### Does this PR introduce a user-facing change?

```release-note
None
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs
None
```

### CODE REVIEW

File: pkg/controller/podautoscaler/replica_calculator_test.go [hunk 1]
```diff
@@ -348,6 +348,7 @@ func (tc *replicaCalcTestCase) runTest(t *testing.T) {
 	stop := make(chan struct{})
 	defer close(stop)
 	informerFactory.Start(stop)
+	informerFactory.WaitForCacheSync(stop)
```
Comment (jackfrancis) [inline pkg/controller/podautoscaler/replica_calculator_test.go:351]: According to @googs1025's reference examples these should be changed to a `ctx.Done()` or equivalent, and we don't need to be creating and passing in a channel.

Ref:

- https://github.com/kubernetes/kubernetes/pull/126228/files#diff-4caf67d968fc1971b19326934df3dd450b6ca42fbd82031bf343d6480d99e7c6L6449

Comment (googs1025) [reply pkg/controller/podautoscaler/replica_calculator_test.go:351]: That makes sense to me. 👍

Comment (omerap12) [reply pkg/controller/podautoscaler/replica_calculator_test.go:351]: No problem , I'll adjust

Comment (aojea) [reply pkg/controller/podautoscaler/replica_calculator_test.go:351]: you don't need another wait for cache, do you?

just below you have 
```
	if !cache.WaitForNamedCacheSync("HPA", stop, informer.Informer().HasSynced) {
		return
	}
```

for the context comment, the other example already had a context, this does not have it so the channel usage loooks corrcet, what we want is to not leack goroutines, and this serves that purpose

Comment (omerap12) [reply pkg/controller/podautoscaler/replica_calculator_test.go:351]: I understand. So closing this PR :)

**Pull request closed without merging**